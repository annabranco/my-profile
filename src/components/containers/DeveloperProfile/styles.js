import styled from 'styled-components';
import { rgba } from 'polished';
import Sea from '../../../images/bg/sea.gif';

export const SectionDeveloper = styled.section`
  position: relative;
  background-image: linear-gradient(
      ${rgba('#ffffff', 0.92)},
      ${rgba('#f0f1fc', 0.93)},
      ${rgba('#e6effb', 0.98)}
    ),
    url(${Sea});
  background-position: center;
  height: auto;
  overflow: hidden;
  padding-top: 50px;
`;

export const DeveloperInfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: auto;
`;
