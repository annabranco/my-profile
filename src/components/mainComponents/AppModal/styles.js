/* eslint-disable no-unused-vars */

import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorBlack,
  colorBlueDark,
  colorWhite,
  fontTitleAlt
} from '../../../styles/theme';

export const CloseButton = styled.p`
  z-index: 100;
  position: fixed;
  top: 0;
  right: 0;
  z-index: 100;
  color: red;
  color: ${({ colors }) => colors.terciary};
  background: ${({ colors }) => colors && colors.primary};
  cursor: pointer;
  border-radius: 5px;
  padding: 5px 10px;

  &:hover {
    box-shadow: 0 0 2px 2px white;
  }

  ${({ text }) =>
    text &&
    css`
      &::before {
        content: 'x';
        text-transform: uppercase;
        font-weight: 700;
        font-family: ${fontTitleAlt};
        font-size: 1rem;
      }
    `};

  @media all and (min-width: 768px) {
    ${({ colors, text }) =>
      text &&
      css`
        top: 20px;
        right: 50px;
        border: ${`2px solid ${colors.terciary}`};
        background: none;

        &::before {
          content: '${text}';
          text-transform: uppercase;
          font-weight: 700;
        }
      `};
  }
`;
CloseButton.displayName = 'CloseButton';

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: ${`${window.innerHeight}px`};
  overflow: hidden;
  width: 100vw;
  z-index: 100;
  background-image: linear-gradient(
    to bottom,
    ${rgba(colorBlueDark, 0.9)},
    ${rgba(colorBlueDark, 0.9)},
    ${rgba(colorBlack, 1)}
  );
  padding: 0;
`;
ModalOverlay.displayName = 'ModalOverlay';
