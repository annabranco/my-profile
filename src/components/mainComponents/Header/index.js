import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { bool } from 'prop-types';
import appInfo from '../../../../package.json';
import {
  allLanguagesSelector,
  currentLanguageSelector,
  headerTitleSelector
} from '../../../redux/selectors';
import { onChangeLanguage } from '../../core/LanguagesModal';
import {
  AppTitle,
  Flag,
  HeaderArea,
  LanguagesWrapper,
  Version
} from './styles';

const Header = ({ hideForever }) => {
  const APP_VERSION = appInfo.version;
  const languages = useSelector(allLanguagesSelector);
  const languageSelected = useSelector(currentLanguageSelector);
  const title = useSelector(headerTitleSelector);
  const dispatch = useDispatch();

  const onCLickFlag = event =>
    onChangeLanguage(
      languageSelected,
      event.currentTarget.lang,
      hideForever,
      dispatch
    );

  return (
    <HeaderArea>
      <LanguagesWrapper>
        {languages
          .filter(item => item.active)
          .map(item => (
            <Flag
              aria-label={item.language}
              key={item.languageCode}
              lang={item.languageCode}
              languageSelected={languageSelected}
              onClick={onCLickFlag}
              role="button"
              tabIndex={0}
            >
              <img
                alt=""
                src={`https://www.countryflags.io/${item.flagCode}/flat/16.png`}
              />
            </Flag>
          ))}
      </LanguagesWrapper>
      <AppTitle>{title}</AppTitle>
      <Version>{APP_VERSION}</Version>
    </HeaderArea>
  );
};

Header.propTypes = {
  hideForever: bool.isRequired
};

export default Header;
