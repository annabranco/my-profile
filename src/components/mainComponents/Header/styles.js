import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import {
  colorGrayNormal,
  colorWhite,
  fontTitleAlt,
  fontTitle
} from '../../../styles/theme';

const headerInner = css`
  min-width: 250px;
  text-align: center;
`;

const activeFlag = css`
  opacity: 1;
  cursor: default;
`;

export const AppTitle = styled.h2`
  ${headerInner}
  order  : 1;
  height: 20px;
  font-family: ${fontTitleAlt};

  @media all and (min-width: 500px) {
    order: 2;
  }
`;
AppTitle.displayName = 'AppTitle';

export const Flag = styled.div`
  display: inline;
  margin: 0 10px;
  outline: none;
  opacity: 0.3;
  cursor: pointer;

  ${props =>
    props.lang === props.languageSelected &&
    css`
      ${activeFlag}
    `}
`;
Flag.displayName = 'Flag';

export const HeaderArea = styled.header`
  z-index: 2;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-wrap: wrap;
  background-image: linear-gradient(
    ${colorGrayNormal},
    ${rgba(colorWhite, 0.6)} 50%
  );
  height: 40px;
  width: 100%;
  padding: 0 80px;

  @media all and (min-width: 500px) {
    flex-wrap: nowrap;
  }
`;
HeaderArea.displayName = 'Header';

export const LanguagesWrapper = styled.div`
  ${headerInner}
  order: 3;
  height: 20px;
  @media all and (min-width: 500px) {
    order: 1;
  }
`;
LanguagesWrapper.displayName = 'LanguagesWrapper';

export const Version = styled.p`
  ${headerInner}
  font-family: ${fontTitle};
  order: 3;
`;
Version.displayName = 'Version';
