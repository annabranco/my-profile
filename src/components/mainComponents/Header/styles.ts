import styled, { css } from 'styled-components';
import { rgba } from 'polished';
import { ILanguageCode } from '../../../types/interfaces';
import {
  colorGrayNormal,
  colorWhite,
  fontTitleAlt,
  fontTitle
} from '../../../styles/theme';

const headerInner = css`
  min-width: 250px;
  text-align: center;
`;

const activeFlag = css`
  opacity: 1;
  cursor: default;
`;

export const AppTitle = styled.h2`
  ${headerInner}
  order  : 1;
  height: 20px;
  font-family: ${fontTitleAlt};
  text-transform: uppercase;
  font-size: 1.2rem;

  @media all and (min-width: 500px) {
    text-transform: unset;
    order: 2;
    font-size: 1rem;
  }
  @media all and (min-width: 2000px) {
    font-size: 1.2rem;
  }
`;
AppTitle.displayName = 'AppTitle';

interface FlagProps extends React.ComponentPropsWithoutRef<'div'> {
  lang?: ILanguageCode;
  languageSelected: ILanguageCode;
}

export const Flag = styled.div<FlagProps>`
  display: inline;
  margin: 0 5px;
  outline: none;
  opacity: 0.3;
  cursor: pointer;
  max-width: 32px;

  ${({ lang, languageSelected }) =>
    lang === languageSelected &&
    css`
      ${activeFlag}
    `}

  @media all and (min-width: 768px) {
    margin: 0 10px;
  }
`;
Flag.displayName = 'Flag';

export const HeaderArea = styled.header`
  z-index: 50;
  position: fixed;
  top: 0;
  left: 0;
  display: flex;
  justify-content: space-around;
  align-items: center;
  flex-wrap: wrap;
  background-image: linear-gradient(
    ${colorGrayNormal},
    ${rgba(colorWhite, 0.6)} 50%
  );
  height: 40px;
  width: 100vw;
  padding: 0 20px;

  @media all and (min-width: 500px) {
    flex-wrap: nowrap;
    padding: 0 80px;
  }
`;
HeaderArea.displayName = 'Header';

export const LanguagesWrapper = styled.div`
  ${headerInner}
  order: 2;
  height: 20px;

  @media all and (min-width: 500px) {
    order: 1;
  }
`;
LanguagesWrapper.displayName = 'LanguagesWrapper';

export const Version = styled.p`
  display: none;
  ${headerInner}
  font-family: ${fontTitle};
  order: 3;

  @media all and (min-width: 768px) {
    display: block;
  }
`;
Version.displayName = 'Version';
