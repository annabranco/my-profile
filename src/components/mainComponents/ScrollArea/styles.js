import styled, { css } from 'styled-components';

export const ScrollAreaWrapper = styled.main`
  height: ${`${window.innerHeight}px`};
  overflow: hidden;
  position: absolute;

  ${props =>
    props.langModalVisible &&
    css`
      overflow-y: hidden;
    `}

  @media all and (min-width: 768px) {
    position: unset;
    overflow-y: scroll;
  }
`;
ScrollAreaWrapper.displayName = 'Scroll Area';
