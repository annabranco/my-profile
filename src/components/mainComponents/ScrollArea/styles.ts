import styled, { css } from 'styled-components';

interface ScrollAreaWrapperProps
  extends React.ComponentPropsWithoutRef<'main'> {
  langModalVisible: boolean;
  isSeabedElementOpened: boolean;
}

export const ScrollAreaWrapper = styled.main<ScrollAreaWrapperProps>`
  height: ${`${window.innerHeight}px`};
  overflow: hidden;
  position: absolute;
  scroll-snap-type: y proximity;

  ${({ langModalVisible }) =>
    langModalVisible &&
    css`
      overflow-y: hidden;
    `}

  @media all and (min-width: 768px) {
    position: unset;
    overflow-y: ${({ isSeabedElementOpened }) =>
      isSeabedElementOpened ? 'hidden' : 'scroll'};
  }
`;
ScrollAreaWrapper.displayName = 'Scroll Area';

interface ScrollSectionElementProps
  extends React.ComponentPropsWithoutRef<'div'> {
  last: boolean;
}

export const ScrollSectionElement = styled.div<ScrollSectionElementProps>`
  scroll-snap-align: ${({ last }) => last && 'end'};
`;
ScrollSectionElement.displayName = 'ScrollSectionElement';
