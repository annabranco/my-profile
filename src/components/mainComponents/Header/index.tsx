import { ReactElement } from 'react';
import * as React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { bool } from 'prop-types';
import {
  ILanguageCode,
  ILanguage,
  ISectionsText
} from '../../../types/interfaces';
import appInfo from '../../../../package.json';
import {
  allLanguagesSelector,
  currentLanguageSelector,
  currentSecionSelector,
  headerTitleSelector,
  sectionsTextsSelector
} from '../../../redux/selectors';
import { isDesktop } from '../../../utils/device';
import { getFlagURL } from '../../../utils/icons';
import { onChangeLanguage } from '../../elements/LanguagesModal';
import { INFO_PAGE_SECTION } from '../../../constants';
import { CountryFlag } from '../../sections/Experiences/styles';
import {
  AppTitle,
  Flag,
  HeaderArea,
  LanguagesWrapper,
  Version
} from './styles';

interface Props {
  hideForever: boolean;
  isSeabedElementOpened: boolean;
}

const Header = ({
  hideForever,
  isSeabedElementOpened
}: Props): ReactElement => {
  const APP_VERSION: string = appInfo.version;
  const languages: ILanguage[] = useSelector(allLanguagesSelector);
  const languageSelected: ILanguageCode = useSelector(currentLanguageSelector);
  const title: string = useSelector(headerTitleSelector);
  const currentSection: string = useSelector(currentSecionSelector);
  const sections: ISectionsText = useSelector(sectionsTextsSelector);

  const dispatch = useDispatch();

  const onCLickFlag = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ): void =>
    onChangeLanguage(
      languageSelected,
      event.currentTarget.lang,
      hideForever,
      dispatch
    );

  const getTitle = (): string => {
    if (isDesktop || currentSection === INFO_PAGE_SECTION) {
      return title;
    }
    return sections[currentSection];
  };

  return (
    <>
      {!isSeabedElementOpened && (
        <HeaderArea>
          <LanguagesWrapper>
            {languages
              .filter(item => item.active)
              .map(item => (
                <Flag
                  aria-label={item.language}
                  data-e2e-id={`header-flag_${item.flagCode}`}
                  key={item.languageCode}
                  lang={item.languageCode}
                  languageSelected={languageSelected}
                  onClick={onCLickFlag}
                  role="button"
                  tabIndex={0}
                >
                  <CountryFlag
                    alt=""
                    src={getFlagURL({
                      country: item.flagCode,
                      style: 'flat3d',
                      size: 'small'
                    })}
                  />
                </Flag>
              ))}
          </LanguagesWrapper>
          <AppTitle data-e2e-id="appTitle">{getTitle()}</AppTitle>
          <Version>{APP_VERSION}</Version>
        </HeaderArea>
      )}
    </>
  );
};

Header.propTypes = {
  hideForever: bool.isRequired,
  isSeabedElementOpened: bool.isRequired
};

export default Header;
