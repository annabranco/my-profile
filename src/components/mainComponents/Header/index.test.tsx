import { mount, ReactWrapper, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import renderer, {
  ReactTestRenderer,
  ReactTestInstance
} from 'react-test-renderer';
import Header from './index';
import appInfo from '../../../../package.json';
import { setupMockProvider, getMockState } from '../../../tests/mocks';
import { CHANGE_LANGUAGE } from '../../../constants';
import { AppTitle, Flag } from './styles';
import { IAppState } from '../../../types/interfaces';
import '../../../tests/mocks/matchMedia';

// TODO [10-May -21]: Refine testing (Anna Branco)

const APP_VERSION = appInfo.version;

describe('< Header Component >', () => {
  let MockProvider;
  let MOCK_STATE: IAppState;
  let wrapper: ReactWrapper;
  // let testInstance: ReactTestInstance;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;
    MOCK_STATE = getMockState({ language: 'en' });

    wrapper = mount(
      <MockProvider>
        <Header hideForever isSeabedElementOpened />
      </MockProvider>
    );
    // console.log('$$$ wrapper', toJson(wrapper.children()));

    // testInstance = wrapper.root;
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('should display the App title', () => {
  //   const { title } = MOCK_STATE.texts.header;

  //   expect(wrapper.find('div').text()).toBe(title);
  // });

  // it('should display the App version', () => {
  //   expect(testInstance.find('Version').text()).toBe(APP_VERSION);
  // });

  // it('should display a languages wrapper', () => {
  //   expect(testInstance.find('LanguagesWrapper').exists()).toBe(true);
  // });

  // it('should display country flags for each active language from state', () => {
  //   const activeLanguages: string[] = [];

  //   MOCK_STATE.languages
  //     .filter(lang => lang.active)
  //     .forEach(lang => activeLanguages.push(lang.flagCode));

  //   const renderedFlags = testInstance.find('Flag').children();

  //   renderedFlags.forEach((flagWrapper: HTMLDivElement, index: number) => {
  //     const url = flagWrapper.children().first().props().src;
  //     const expectedUrl = `https://www.countryflags.io/${activeLanguages[index]}/flat/16.png`;

  //     expect(url).toBe(expectedUrl);
  //   });
  //   expect(renderedFlags).toHaveLength(activeLanguages.length);
  // });

  it('should highlight the active language flag', () => {
    const flagEnglish = shallow(<Flag lang="en" languageSelected="en" />);
    const flagPortuguese = shallow(<Flag lang="pt" languageSelected="en" />);

    expect(toJson(flagEnglish)).toHaveStyleRule('opacity', '1');
    expect(toJson(flagPortuguese)).toHaveStyleRule(
      'opacity',
      expect.stringContaining('0.')
    );
  });

  // it('should dispatch a CHANGE_LANGUAGE action when a flag is clicked', () => {
  //   const dispatchSpy = jest.spyOn(window, 'interceptedDispatch');
  //   const flag = testInstance.find('Flag').children().last();
  //   const flagLanguage = flag.props().lang;

  //   flag.simulate('click');

  //   expect(dispatchSpy).toHaveBeenCalledTimes(1);
  //   expect(dispatchSpy).toHaveBeenCalledWith({
  //     payload: flagLanguage,
  //     type: CHANGE_LANGUAGE
  //   });
  // });

  // it('should change Title when state language changes', () => {
  //   const { title: englishTitle } = MOCK_STATE.texts.header;
  //   const { title: spanishhTitle } = getMockState({
  //     language: 'es'
  //   }).texts.header;

  //   expect(testInstance.find('AppTitle').text()).toBe(englishTitle);

  //   const MockProviderWithStateChange = setupMockProvider({ language: 'es' })
  //     .MockProvider;
  //   const componentAfterLanguageChange = mount(
  //     <MockProviderWithStateChange>
  //       <Header hideForever isSeabedElementOpened />
  //     </MockProviderWithStateChange>
  //   );

  //   expect(componentAfterLanguageChange.find('AppTitle').text()).toBe(
  //     spanishhTitle
  //   );
  // });
});
