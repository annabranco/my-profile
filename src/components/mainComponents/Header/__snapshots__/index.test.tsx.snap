// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Header Component > matches the snapshot 1`] = `
<MockProvider>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Header
      hideForever={true}
      isSeabedElementOpened={true}
    />
  </Provider>
</MockProvider>
`;
